### ISSUE TITLE
ModifyCollectionInEnhancedForLoop is triggered for ConcurrentMap

### ISSUE THREAD

**Issue by ejona86** [google/error-prone#1767]
### Description of the problem / feature request:

ModifyCollectionInEnhancedForLoop is triggered for ConcurrentMap. While this _could_ be on purpose, as technically ConcurrentHashMap defines the no-throw-ConcurrentModificationException policy, it seems this is accidental in some way as ConcurrentMap is part of `java.util.concurrent` which is skipped by the check.

### Bugs: what's the simplest, easiest way to reproduce this bug?

We have the code:
```
  private final ConcurrentMap<Locality, ClientLoadCounter> localityLoadCounters;
...
    for (Map.Entry<Locality, ClientLoadCounter> entry : localityLoadCounters.entrySet()) {
        ...
        localityLoadCounters.remove(entry.getKey());
```

Which causes a ModifyCollectionInEnhancedForLoop warning:
```
grpc-java/xds/src/main/java/io/grpc/xds/LoadStatsStoreImpl.java:94: warning: [ModifyCollectionInEnhancedForLoop] Modifying a collection while iterating over it in a loop may cause a ConcurrentModificationException to be thrown or lead to undefined behavior.
        localityLoadCounters.remove(entry.getKey());
                                   ^
    (see https://errorprone.info/bugpattern/ModifyCollectionInEnhancedForLoop)
```

Swapping from ConcurrentMap to ConcurrentHashMap silences the warning.

Looking at the code, there does not seem to be any ConcurrentMap vs ConcurrentHashMap special-casing, just a large special-case for `java.util.concurrent`:
https://github.com/google/error-prone/blob/55a5956c6f683b9070e49553417f3a398d6bdf31/core/src/main/java/com/google/errorprone/bugpatterns/ModifyCollectionInEnhancedForLoop.java#L76-L85

It's unclear how ConcurrentMap triggers the warning.

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

No.
